tensorflow/python/training/evaluation.py

```diff
@@ -106,7 +108,8 @@ def after_run(self, run_context, run_values):
       if self._num_evals is None:
         logging.info('Evaluation [%d]', evals_completed)
       else:
-        logging.info('Evaluation [%d/%d]', evals_completed, self._num_evals)
+        if (evals_completed % self._log_frequency) == 1:
+          logging.info('Evaluation [%d/%d]', evals_completed, self._num_evals)
```

Comment:
Here you need one more condition

if ((evals_completed % self._log_frequency) == 1 or
    self._num_evals == evals_completed):
  logging.info('Evaluation [%d/%d]', evals_completed, self._num_evals)

It is very important from usability perspective to print the final one.

Reply:
Thanks for being so patient. Here's a couple of examples where we are at now:

With 390 steps:
```
INFO:tensorflow:Evaluation [39/390]
INFO:tensorflow:Evaluation [78/390]
INFO:tensorflow:Evaluation [117/390]
INFO:tensorflow:Evaluation [156/390]
INFO:tensorflow:Evaluation [195/390]
INFO:tensorflow:Evaluation [234/390]
INFO:tensorflow:Evaluation [273/390]
INFO:tensorflow:Evaluation [312/390]
INFO:tensorflow:Evaluation [351/390]
INFO:tensorflow:Evaluation [390/390]
```

With 25 steps:
```
INFO:tensorflow:Evaluation [2/25]
INFO:tensorflow:Evaluation [4/25]
INFO:tensorflow:Evaluation [6/25]
INFO:tensorflow:Evaluation [8/25]
INFO:tensorflow:Evaluation [10/25]
INFO:tensorflow:Evaluation [12/25]
INFO:tensorflow:Evaluation [14/25]
INFO:tensorflow:Evaluation [16/25]
INFO:tensorflow:Evaluation [18/25]
INFO:tensorflow:Evaluation [20/25]
INFO:tensorflow:Evaluation [22/25]
INFO:tensorflow:Evaluation [24/25]
INFO:tensorflow:Evaluation [25/25]
```

With 10 steps
```
INFO:tensorflow:Evaluation [1/10]
INFO:tensorflow:Evaluation [2/10]
INFO:tensorflow:Evaluation [3/10]
INFO:tensorflow:Evaluation [4/10]
INFO:tensorflow:Evaluation [5/10]
INFO:tensorflow:Evaluation [6/10]
INFO:tensorflow:Evaluation [7/10]
INFO:tensorflow:Evaluation [8/10]
INFO:tensorflow:Evaluation [9/10]
INFO:tensorflow:Evaluation [10/10]
```

Seems reasonable to me, although it's not always exactly 10 outputs, because of the rounding / truncation step.